tutorial.md

```diff
@@ -0,0 +1,1212 @@
+achieved with the `additionalKeysP` function from the
+[XMonad.Util.EZConfig] module—luckily we already have it imported!  Our
+config file, starting with `main`, now looks like:
+
+``` haskell
+  main :: IO ()
+  main = xmonad $ def
+      { modMask = mod4Mask  -- Rebind Mod to the Super key
+      }
+    `additionalKeysP`
+      [ ("M-S-z", spawn "xscreensaver-command -lock")
+      , ("M-S-=", unGrab *> spawn "scrot -s"        )
+      , ("M-]"  , spawn "firefox"                   )
+      ]
+```
+
+That syntax look familiar?
+
+You can find the names for special keys, like `Print` or the `F`-keys,
+in the [XMonad.Util.EZConfig] documentation.
+
+We will cover setting up the screensaver later in this tutorial.
+
+The `unGrab` before running the `scrot -s` command tells xmonad to
+release its keyboard grab before `scrot -s` tries to grab the keyboard
+itself.  The little `*>` operator essentially just sequences two
+functions, i.e. `f *> g` says
+
+  > first do `f` and then, discarding any result that `f` may have given
+  > me, do `g`.
+
+Do note that you may need to install `scrot` if you don't have it on
+your system already.
+
+What if we wanted to augment our xmonad experience just a little more?
+We already have `xmonad-contrib`, which means endless possibilities!
+Say we want to add a three column layout to our layouts and also magnify
+focused stack windows, making it more useful on smaller screens.
+
+We start by visiting the documentation for [XMonad.Layout.ThreeColumns].
+The very first sentence of the `Usage` section tells us exactly what we
+want to start with:
+
+  > You can use this module with the following in your `~/.xmonad/xmonad.hs`:
+  >
+  > `import XMonad.Layout.ThreeColumns`
+
+Ignoring the part about where exactly our `xmonad.hs` is (we have opted
+to put it into `~/.config/xmonad/xmonad.hs`, remember?) we can follow
+that documentation word for word.  Let's add
+
+``` haskell
+  import XMonad.Layout.ThreeColumns
+```
+
+to the top of our configuration file.  Most modules have a lot of
+accompanying text and usage examples in them—so while the type
+signatures may seem scary, don't be afraid to look up the
+[xmonad-contrib documentation] on hackage!
+
+Next we just need to tell xmonad that we want to use that particular
+layout.  To do this, there is the `layoutHook`.  Let's use the default
+layout as a base:
+
+``` haskell
+  myLayout = tiled ||| Mirror tiled ||| Full
+    where
+      tiled   = Tall nmaster delta ratio
+      nmaster = 1      -- Default number of windows in the master pane
+      ratio   = 1/2    -- Default proportion of screen occupied by master pane
+      delta   = 3/100  -- Percent of screen to increment by when resizing panes
+```
+
+The so-called `where`-clause above simply consists of local declarations
+that might clutter things up where they all declared at the top-level
+like this
+
+``` haskell
+  myLayout = Tall 1 (1/2) (3/100) ||| Mirror (Tall 1 (1/2) (3/100)) ||| Full
```

Comment:
A minor nitpick: this code snippet sees the `ratio` and `delta` parameters switched (delta is 1/2 and ratio is 3/100) for both Tall configurations.

This should really read:
```haskell
myLayout = Tall 1 (3/100) (1/2) ||| Mirror (Tall 1 (3/100) (1/2)) ||| Full
```

Reply:
Well spotted, thanks!